numpy/lib/_npyio_impl.py

```diff
@@ -948,8 +948,8 @@ def _read(fname, *, delimiter=',', comment='#', quote='"',
     dtype = np.dtype(dtype)
 
     read_dtype_via_object_chunks = None
-    if dtype.kind in 'SUM' and (
-            dtype == "S0" or dtype == "U0" or dtype == "M8" or dtype == 'm8'):
+    if dtype.kind in 'SUM' and dtype in {
+            np.dtype("S0"), np.dtype("U0"), np.dtype("M8"), np.dtype('m8')}:
```

Comment:
```suggestion
            np.dtype("S0"), np.dtype("U0"), np.dtype("M8"), np.dtype("m8")}:
```
(or the other way around)